hibernate-core/src/main/java/org/hibernate/query/internal/QueryInterpretationCacheStandardImpl.java

```diff
@@ -200,7 +205,7 @@ public void close() {
 		queryPlanCache.clear();
 	}
 
-	private static final class HqlInterpretationCacheKey {
+	public static final class HqlInterpretationCacheKey {
```

Comment:
Tangential: we should convert this `class` to a `record`.

Reply:
Should/could... but sure.

Reply:
Hm, actually, given that it defines a specialized `equals`, does it really make sense though?

Reply:
Ah, you are right.  I did not look closely at it, but you are right.